crates/sequencer/src/serde/state.rs

```diff
@@ -0,0 +1,156 @@
+use std::{fs, path::PathBuf};
+
+use blockifier::state::cached_state::ContractStorageKey;
+use rustc_hash::FxHashMap;
+use serde::{Deserialize, Serialize};
+use starknet_api::{
+    core::{ClassHash, CompiledClassHash, ContractAddress, Nonce},
+    hash::StarkFelt,
+};
+use thiserror::Error;
+
+use crate::state::State;
+
+use super::contract::SerializableContractClass;
+
+/// TODO: wrap the underlying errors in them
+#[derive(Error, Debug)]
+pub enum SerializationError {
```

Comment:
I think we can rework this a little to use more the syntax from thiserror examples?
```
#[derive(Error, Debug)]
pub enum MyError {
    Io {
        #[from]
        source: io::Error,
        backtrace: Backtrace,
    },
}
```
[source](https://docs.rs/thiserror/latest/thiserror/#details)